networkx/drawing/tests/test_pylab.py

```diff
@@ -1152,6 +1152,36 @@ def test_draw_nodes_missing_node_from_position():
         nx.draw_networkx_nodes(G, pos)
 
 
+def test_draw_networkx_nodes_node_shape_list_with_scalar_color(subplots):
+    """Ensure draw_networkx_nodes works when node_shape is a Python list.
+
+    This covers the case where node_shape is a sequence (list) and node_color
+    is a single scalar color, which should be supported.
+    """
+    fig, ax = subplots
+
+    G = nx.Graph()
+    G.add_nodes_from(range(5))
+    pos = nx.random_layout(G)
+
+    shapes = ["o", "^", "o", "^", "o"]
+
+    nodes = nx.draw_networkx_nodes(
+        G,
+        pos,
+        node_color="red",  # scalar color (supported)
+        node_shape=shapes,  # list of shapes – this used to be buggy
```

Comment:
I don't think this is quite right - this is already allowed and will pass on `main`. The unit test should fail on `main` and pass with the fix. The easiest way to do this is to adhere more closely to the example in the OP of the issue, which uses `colors=["red"]*5` (i.e. a list of scalars).

Reply:
~~whoops... missed that!
Yes -- this test should be using a list for each of node_color and node_shape.
Good catch!~~

Reply:
Whoops again -- 
I think the test "as is" would fail on main right now. It works for `node_shape` being an array, but not a list.